Platform/CommonBoardPkg/Library/BoardSupportLib/BoardSupportLib.c

```diff
@@ -533,3 +536,51 @@ GetVbtAddress ()
 
   return PcdGet32(PcdGraphicsVbtAddress);
 }
+
+/**
+  Patch VBT to use a fixed display mode with the required resolution.
+
+  @param[in]  VbtBuf    VBT binary buffer in memory to be patched.
+  @param[in]  Xres      Requested mode X resolution.
+  @param[in]  Yres      Requested mode Y resolution.
+
+  @retval     EFI_SUCCESS        Fixed mode block in VBT has been patched to the required mode.
+              EFI_NOT_FOUND      Could not find fixed mode block in VBT.
+**/
+EFI_STATUS
+EFIAPI
+SetVbtFixedMode (
+  IN  UINT8     *VbtBuf,
+  IN  UINT32     Xres,
+  IN  UINT32     Yres
+  )
+{
+  EFI_STATUS                Status;
+  VBT_HEADER               *VbtHdr;
+  VBT_BIOS_DATA_HEADER     *BiosDataHdr;
+  VBT_BLOCK_COMMON_HEADER  *BlkHdr;
+  BLOCK51_FIXED_MODE_SET   *ModeBlk;
+  UINT32                    Offset;
+
+  Status = EFI_NOT_FOUND;
+  if ((VbtBuf != NULL) && (*(UINT32 *)VbtBuf == VBT_SIGNATURE)) {
+    VbtHdr = (VBT_HEADER *)VbtBuf;
+    Offset = VbtHdr->Bios_Data_Offset;
+    BiosDataHdr = (VBT_BIOS_DATA_HEADER *)(VbtBuf + Offset);
+    Offset = Offset + BiosDataHdr->BDB_Header_Size;
+    while (Offset < BiosDataHdr->BDB_Size) {
+      BlkHdr = (VBT_BLOCK_COMMON_HEADER *)(VbtBuf + Offset);
+      // BlockId 51 is for fixed mode set
+      if (BlkHdr->BlockId == 51) {
+        ModeBlk = (BLOCK51_FIXED_MODE_SET *) BlkHdr;
+        ModeBlk->FeatureEnable = 1;
+        ModeBlk->XRes = Xres;
+        ModeBlk->YRes = Yres;
+        Status = EFI_SUCCESS;
+        break;
+      }
+      Offset += (BlkHdr->BlockSize + sizeof(VBT_BLOCK_COMMON_HEADER));

```

Comment:
If VBT_BLOCK_COMMON_HEADER is added to Offset here, in the loop line 572 it is confused (VbtBuf + Offset) is still a pointer of VBT_BLOCK_COMMON_HEADER. Could you please double check?

Reply:
It points to next block header.  What is the problem here?

Reply:
if (VbtBuf + Offset) point to a block header in Line 572, then "Offset += BlkHdr->BlockSize" should point to next block header. here why it add sizeof(VBT_BLOCK_COMMON_HEADER)?

Reply:
Per spec, the BlockSize indicated in the header does not include the VBT_BLOCK_COMMON_HEADER itself.

Reply:
Got it.